src/www/ui/showjobs.php

```diff
@@ -146,7 +134,7 @@ public function Output()
       $jq_pk = GetParm("jobid",PARM_INTEGER);
       $action = GetParm("action",PARM_STRING);
       $uploadPk = GetParm("upload",PARM_INTEGER);
-      if (!empty($uploadPk) && !$this->uploadDao->isEditable($uploadPk, Auth::getGroupId())){
+      if (!(empty($uploadPk) || $uploadPk === -1) && !$this->uploadDao->isEditable($uploadPk, Auth::getGroupId())){
```

Comment:
This is a security issue that was also wrong before: all users with userlevel >= PLUGIN_DB_WRITE can trigger actions (e.g. cancel) on jobs for uploads, even if their group is not permitted to edit the upload